File: front/frontclient/src/app/movement.ts
import { FIGURE } from "./figure";
import { PLAYER } from "./player";

export class Movement {
    id: number | undefined;
    player: PLAYER | null;
    figure: FIGURE | null;

    constructor() {
        this.id = undefined;
        this.player = null;
        this.figure = null;
    }
}

File: src/main/kotlin/games/rockpaperscissors/model/Movement.kt
package games.rockpaperscissors.model

import jakarta.persistence.*


@Entity
@Table(name = "movement")
data class Movement(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null,
    @Column(nullable = false)
    val player: Player,
    @Column(nullable = false)
    val figure: Figure,
    @Column
    val game_id: Int? = null
)